src/hardware/ipx.cpp

```diff
@@ -29,27 +30,38 @@
 #include "programs.h"
 #include "pic.h"
 
+constexpr int SocketWaitMs = 1000;
+
+static std::thread ipx_client_thread = {};
+
 #define SOCKTABLESIZE	150 // DOS IPX driver was limited to 150 open sockets
 
+constexpr int PacketQueueSize = 10;
+
 struct ipxnetaddr {
 	Uint8 netnum[4];   // Both are big endian
 	Uint8 netnode[6];
 } localIpxAddr;
 
+struct IpxPacket {
+	UDPpacket udp;
+	uint8_t recv_buffer[IPXBUFFERSIZE];
+	std::atomic_bool available;
+};
```

Comment:
Please add default values. All nulls are fine, otherwise we'll get random garbage on init depending on memory layout and that's a bitch to debug.

So:

```cpp
struct IpxPacket {
    UDPpacket udp = {};
    uint8_t recv_buffer[IPXBUFFERSIZE] = {};
    std::atomic_bool available = {};
}; 
```

Also, can you please use a `std::vector` instead of raw arrays whenever possible in new code. C arrays are error prone to deal with.